langroid/agent/chat_agent.py

```diff
@@ -179,8 +187,16 @@ def __init__(
         self.output_format: Optional[type[ToolMessage | BaseModel]] = None
 
         self.saved_requests_and_tool_setings = self._requests_and_tool_settings()
+        # Tracks if we have enabled use of an output format
```

Comment:
Needs better explanation, took me a while to grok it; suggest this  (maybe you need to edit it if not quite right):

When this is not None and equals a tool msg T, this means that 
(a) T has been set as the `output_format` of this agent, AND 
(b) T has been "enabled for use" ONLY for enforcing this output format.
Thus when the `output_format` changes, T will need to be "disabled for use".